### ISSUE TITLE
Request feature: Indentition Check

### ISSUE THREAD

**Issue by stczwd** [koalaman/shellcheck#2463]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:
```sh
#!/bin/bash
echo "this is a test"
                echo "thjsadfa"
                if [[ 1 -lt 2 ]]; then
                                                echo "ok"
                                                fi
```

#### Here's what shellcheck currently says:
```
```

#### Here's what I wanted or expected to see:
Based on [Google Shell Style Guide](https://google.github.io/styleguide/shellguide.html#s5.1-indentation),  we should use 2 spaces and no tabs for indentition. Besides, wrong indentitions should be check for code style.

**Comment by SuperSandro2000** [comment]
You can use something like [shfmt](https://github.com/mvdan/sh) to achieve that.

**Comment by stczwd** [comment]
> You can use something like [shfmt](https://github.com/mvdan/sh) to achieve that.

Any possible to support indentition check in `shellcheck`? So we can use one tool to check shell style.

**Comment by koalaman** [comment]
I see how that would be convenient, but ShellCheck currently does not really track formatting like indentation so it's not a trivial addition, and since `shfmt` already does such a great job at it it doesn't make sense to duplicate that effort.
